mongo/cursor_test.go

```diff
@@ -102,37 +102,69 @@ func TestCursor(t *testing.T) {
 	t.Run("TestAll", func(t *testing.T) {
 		t.Run("errors if argument is not pointer to slice", func(t *testing.T) {
 			cursor, err := newCursor(newTestBatchCursor(1, 5), nil, nil)
-			assert.Nil(t, err, "newCursor error: %v", err)
+			assert.NoError(t, err, "newCursor error: %v", err)
```

Comment:
If we are checking an error can we use `require.NoError`? It seems there's no point in continuing the test otherwise. Additionally, we don't have to include the error in the message since the assertion will provide it:

```go
require.NoError(t, err, "failed to create cursor")
```